apps/frontend/src/pages/Build/BuildInfoDialog.tsx

```diff
@@ -0,0 +1,100 @@
+import { useParams } from "react-router-dom";
+import { BuildType } from "@/gql/graphql";
+
+import {
+  Dialog,
+  DialogBody,
+  DialogDismiss,
+  DialogFooter,
+  DialogText,
+  DialogTitle,
+  useDialogState,
+} from "@/ui/Dialog";
+import { Code } from "@/ui/Code";
+import { Link } from "@/ui/Link";
+
+const useDialogContent = (
+  buildType: BuildType | null | undefined,
+  referenceBranch: string | null | undefined,
+) => {
+  const { accountSlug, projectName } = useParams();
+
+  switch (buildType) {
+    case BuildType.Orphan:
+      return (
+        <>
+          <DialogText>
+            This build is <span className="font-semibold">orphan</span> because
+            there is no baseline for screenshots comparisons. To start visual
+            testing, create a{" "}
+            <span className="font-semibold">reference build</span> to get a
+            baseline and rebase your branch.
+          </DialogText>
+
+          <DialogText className="mt-1 mb-1">
+            <span className="font-semibold">
+              How to create a reference build?
+            </span>{" "}
+            Run Argos on reference branch
+            {referenceBranch ? (
+              <span>
+                {" "}
+                <Code>{referenceBranch}</Code>
+              </span>
+            ) : (
+              ""
+            )}
+            .
+          </DialogText>
+          <DialogText className="text-xs text-low mt-1">
+            Note: You can change the reference branch on{" "}
+            <Link tabIndex={-1} to={`/${accountSlug}/${projectName}/settings`}>
+              project's settings
+            </Link>
+            .
+          </DialogText>
+
+          <DialogText className="mt-1">
+            <span className="font-semibold">
+              How to keep your baseline updated?
+            </span>{" "}
+            Configure your CI pipeline to run Argos when the reference branch is
+            updated.
+          </DialogText>
+        </>
+      );
+
+    default:
+      return null;
+  }
+};
+
+export const BuildInfoDialog = ({
```

Comment:
It's over engineered, just call it `BuildOrphanDialog` and avoid displaying it you are not on an orphan build. You made it generic but we will display it only on orphan build. It makes the code complicated for nothing.

Reply:
We will not use a dialog to display it, so no not needed. But you are right, I have something to do on ref builds.